api/view_only_links/serializers.py

```diff
@@ -0,0 +1,155 @@
+
+    nodes = RelationshipField(
+        related_view='view-only-links:view-only-link-nodes',
+        related_view_kwargs={'link_id': '<_id>'},
+        self_view='view-only-links:view-only-link-nodes-relationships',
+        self_view_kwargs={'link_id': '<_id>'}
+    )
+
+    creator = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<creator._id>'},
+    )
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse(
+            'nodes:node-view-only-link-detail',
+            kwargs={
+                'link_id': obj._id
+            }
+        )
+
+    class Meta:
+        type_ = 'view-only-links'
+
+
+class VOLNode(JSONAPIRelationshipSerializer):
+    id = ser.CharField(source='_id')
+
+    class Meta:
+        type_ = 'nodes'
+
+
+class ViewOnlyLinkNodesSerializer(ser.Serializer):
+    data = ser.ListField(child=VOLNode())
+    links = LinksField({
+        'self': 'get_self_url',
+    })
+
+    def get_self_url(self, obj):
+        return absolute_reverse(
+            'view-only-links:view-only-link-nodes',
+            kwargs={
+                'link_id': obj['self']._id
+            }
+        )
+
+    def make_instance_obj(self, obj):
+        return {
+            'data': obj.nodes,
+            'self': obj
+        }
+
+    def get_nodes_to_add_remove(self, nodes, new_nodes):
+        diff = relationship_diff(
+            current_items={node._id: node for node in nodes},
+            new_items={node['_id']: node for node in new_nodes}
+        )
+
+        nodes_to_add = []
+        for node_id in diff['add']:
+            node = Node.load(node_id)
+            if not node:
+                raise NotFound
+            nodes_to_add.append(node)
+
+        return nodes_to_add, diff['remove'].values()
+
+    def get_eligible_nodes(self, nodes, add):
+        return [
+            descendant
+            for node in (nodes + add)
+            for descendant in node.get_descendants_recursive()
+            if descendant._parent_node in (nodes + add)
+        ]
+
+    def create(self, validated_data):
+
+        instance = self.context['view'].get_object()
+        view_only_link = instance['self']
+        nodes = instance['data']
+        user = self.context['request'].user
+        new_nodes = validated_data['data']
+
+        add, remove = self.get_nodes_to_add_remove(
+            nodes=nodes,
+            new_nodes=new_nodes
+        )
+
+        if not len(add):
+            raise RelationshipPostMakesNoChanges
+
+        eligible_nodes = self.get_eligible_nodes(nodes, add)
+
+        for node in add:
+            if not node.has_permission(user, 'admin'):
+                raise PermissionDenied
+            if node not in eligible_nodes:
+                raise ValidationError(detail='The node {0} cannot be affiliated with this VOL because it is not a child of the associated VOL nodes.'.format(node._id))
+            view_only_link.nodes.append(node)
```

Comment:
I talked with product. Go ahead and remove this. Make a VOL with something impossible from the front-end's perspective and make sure it isn't disastrous. I'm okay with the front end displaying it right but not being able to reproduce it, and probably even not letting the front end edit it but only delete it. But it should accurately represent the state of the ticket and should let you delete it.

If it is disastrous, the options are to fix it or to make this set of endpoints dev_only. I'll leave it to your better judgement about which way you want to go, but make a note for QA and we can discuss if any additional tickets will need to be made for the future.